paper_framework_sources/p1898_forward_progress_delegation.md

```diff
@@ -301,76 +311,61 @@ When applied to a `receiver`, if supported, will return a `scheduler` that the `
+For some subexpression `r`, let `R` be a type such that `decltype((r))` is `R`
+The expression `execution::get_scheduler(r)` is expression-equivalent to:
 
- * `R.get_scheduler()` if that expression is valid.
- * Otherwise, `get_scheduler(R)` if that expression is valid with overload resolution performed in a context that includes the declaration
+ * `r.get_scheduler()` if that expression is valid, if `R` satisfies `receiver`
+ * Otherwise, `get_scheduler(r)` if that expression is valid, and if `R` satisfies `receiver`, with overload resolution performed in a context that includes the declaration
  ```
-         template<class R>
-           void get_scheduler(R) = delete;
+      void get_scheduler() = delete;
  ```
+   and that does not include a declaration of `execution::get_scheduler`.
 
 
 ## execution::sync_wait modifications
 ### Summary
-Modifications to `sync_wait` as specified in [@P1897R0].
+Modifications to `sync_wait` as specified in [@P1897R3].
 Rather than simply passing a `receiver` to the passed `sender`, `sync_wait` constructs an execution context on the blocked thread, such that the blocked thread drives that executor and executed enqueued work.
-A `scheduler` representing that execution context will be made available from the `receiver` passed to `submit` on the `sender`, thus making that `receiver` a `scheduler_provider`.
+A `scheduler` representing that execution context will be made available from the `receiver` passed to `connect` on the `sender`, thus making that `receiver` a `scheduler_provider`.
 
-
-### Wording
+### Wording to modify final otherwise clause
  * Otherwise:
    * Constructs an execution context owned by the calling thread such that tasks enqueued into that context will be processed by the calling thread.
-   * Constructs a `scheduler`, `s` representing that execution context.
-   * Constructs a `receiver`, `r` over an implementation-defined synchronization primitive and passes that callback to `execution::submit(S, r)`.
-   * `execution::get_scheduler(r)` will return `s`.   
-   * Waits on the synchronization primitive to block on completion of `S`.
+   * Constructs a `scheduler`, `sch` representing that execution context.
+   * Constructs a `receiver`, `r` over an implementation-defined synchronization primitive and passes that callback to `execution::connect(s, r)` such that:
+     * `r` satisfies `scheduler_provider`.
+     * `execution::get_scheduler(r)` will return `sch`.
+     * If `set_value` is called on `r`, the call to `sync_wait` will return the passed value.
+     * If `set_error` is called on `r`, the call to `sync_wait` will throw the passed value as an exception.
+     * If `set_done` is called on `r`, the call to `sync_wait` will throw a cancellation exception.
+   * Waits on the synchronization primitive to block on completion of `s`, driving the execution context as necessary until `set_value`, `set_error` or `set_done` is called on `r` .
```

Comment:
> If `set_done` is called on `r`, the call to `sync_wait` will throw a cancellation exception.

Please no.
returning an optional<T> is the way to express this IMO. It will encourage a better solution in the language and will not make cancellation inaccessible with fnoexceptions

Reply:
This is the more important question I wanted your feedback on on the other PR. Nobody gave me clear feedback on what the preferred direction was when we discussed this last time around.